mgr/dashboard: Carbonised Notification Header

SundownRises opened the pull request:
# Implement New Notification Panel with Carbon Design System Integration

## Summary
This PR introduces a new notification panel component built with Carbon Design System (CDS) components, providing a  alternative to the existing notification sidebar. The implementation includes a boolean flag system to toggle between the new panel and legacy sidebar.
Also added unread notification counter on bell icon.

## Reference
![Screenshot From 2025-07-07 12-05-41](https://github.com/user-attachments/assets/6f0c0534-c5d3-45a4-aca2-a41f56efc3eb)

## g10 theme
<img width="777" height="286" alt="Screenshot From 2025-07-11 11-39-33" src="https://github.com/user-attachments/assets/2e14fe31-20bc-4a9e-9ffa-0ce4a0ed7163" />


https://github.com/user-attachments/assets/033a0f5f-2475-45ee-a8a2-da017e56faf0



## Boolean Flag System
- Panel behavior: New panel is shown (`useNewPanel: true`)
In the NotificationService - to set the  panel type:
Go to src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts
Find line ~26 where useNewPanel: true is set
Change it to `useNewPanel: false` to use the old sidebar 

- In the NavigationComponent - to control which panel opens when clicking the notification icon:
Go to src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.ts
Find the toggleSidebar() method around line 212
Change the second parameter from true to false  to use the old sidebar 


Fixes: https://tracker.ceph.com/issues/71732

This commit carbonises the notification header






## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)
</details>

afreen23 commented:
The lighter theme matching with current dashboard them would be fine.
Also the effect of dismiss button could be:
https://github.com/user-attachments/assets/b00d7bb3-0f7a-4f20-8ded-46b437828446

SundownRises commented:
> The lighter theme matching with current dashboard them would be fine.
> Also the effect of dismiss button could be:
> 
> https://github.com/user-attachments/assets/b00d7bb3-0f7a-4f20-8ded-46b437828446
> 
> 

Looking into it, will fix it soon

afreen23 commented:
jenkins test make check arm64

afreen23 commented:
jenkins test dashboard

afreen23 commented:
jenkins test dashboard cephadm

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/notification-panel/header/notification-header.component.html:
```diff
@@ -0,0 +1,25 @@
+<div class="notification-header">
+  <div class="notification-header__top">
+    <div class="notification-header__title">
+      <cds-text>Tasks and Notifications</cds-text>
```
```suggestion
      <cds-text i18n>Tasks and Notifications</cds-text>
```

SundownRises replied:
Fixed

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/notification-panel/header/notification-header.component.html:
```diff
@@ -0,0 +1,25 @@
+      <cds-text>Tasks and Notifications</cds-text>
+    </div>
+
+    <button
```
```suggestion
    <button
     i18n
```

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/notification-panel/header/notification-header.component.html:
```diff
@@ -0,0 +1,25 @@
+      [checked]="isMuted"
+      size="sm"
+      (checkedChange)="onToggleMute()"
+      label="Mute notifications"
```
```suggestion
      label="Mute notifications"
      i18n-label
```

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/notification-panel/header/notification-header.component.scss:
```diff
@@ -0,0 +1,56 @@
+    cds-toggle {
+      margin: 0;
+
+      ::ng-deep {
+        .cds--toggle__label-text {
```
Same comment lets avoid ng deep as much as we can as its not a good practice.

SundownRises replied:
I will try to avoid it next time thankyou!! Will keep on refining code later

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/notifications/notifications.component.scss:
```diff
@@ -25,3 +28,28 @@ a {
+
+.notification-count {
+  position: absolute;
+  top: spacing.$spacing-01;
+  right: spacing.$spacing-01;
+  min-width: spacing.$spacing-04;
+  height: spacing.$spacing-04;
+  padding: 0 spacing.$spacing-01;
+  border-radius: spacing.$spacing-02;
+  background-color: $support-error;
+  color: $text-on-color;
+  font-size: spacing.$spacing-04;
+  line-height: spacing.$spacing-04;
```
+1 supperb work on using carbon vars, thanks!

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts:
```diff
@@ -20,12 +20,21 @@ export class NotificationService {
-  // Sidebar observable
-  sidebarSubject = new Subject();
+  // Panel state observable
+  private panelStateSource = new BehaviorSubject<{ isOpen: boolean; useNewPanel: boolean }>({
+    isOpen: false,
+    useNewPanel: true
+  });
+  panelState$ = this.panelStateSource.asObservable();
+
+  // Mute state observable
+  private muteStateSource = new BehaviorSubject<boolean>(false);
+  muteState$ = this.muteStateSource.asObservable();
```
for the whole file arrange obs together and subs together

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts:
```diff
@@ -45,6 +54,11 @@ export class NotificationService {
+
+    // Load mute state from localStorage
+    const isMuted = localStorage.getItem(this.MUTE_KEY) === 'true';
+    this.hideToasties = isMuted;
```
We only need to allow toasts for notifications triggered on create/delete/update. Not for this PR but can be experimented later, lets have a tracker so we dont miss.

SundownRises replied:
https://tracker.ceph.com/issues/72838

afreen23 reviewed (changes requested):
Thanks for all work!

Lets squash all commits as per guidelines.

SundownRises commented:
Addressed all comments in this PR!! Make check passed though am facing error in make check-arm64!! Please help me understand error @afreen23 @Pegonzal

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/notification-panel/header/notification-header.component.html:
```diff
@@ -0,0 +1,27 @@
+      [checked]="isMuted"
+      size="sm"
+      (checkedChange)="onToggleMute()"
+      label="Mute notifications"
+      i18n-label
+      hideLabel="true">
```
why setting label when you want to hide ?

SundownRises replied:
Fixed

Pegonzal reviewed (commented):
Hey @SundownRises could you clarify the scope of this PR to me? 

From my testing I can see the following changes but not sure if  I should be seeing the sidebar as well.

<img width="612" height="196" alt="image" src="https://github.com/user-attachments/assets/8dfa5ef1-2600-4b7e-b311-54ed6e7caa61" />

Also, personally I would like to have visual indicator that the notifications have been muted. So how about changing the icon to:

<img width="288" height="262" alt="image" src="https://github.com/user-attachments/assets/2bde2b0c-210b-49c5-857a-63645ad48fbe" />

afreen23 commented:
From my testing I can see the following changes but not sure if I should be seeing the sidebar as well.

>> Yes, the next PR would bring more changes so what are you seeing seems good now, thanks
